scipy/optimize/tests/test_least_squares.py

```diff
@@ -787,6 +788,87 @@ def test_basic():
     assert_allclose(res.x, 0, atol=1e-10)
 
 
+def test_callback():
+    # test that callback function works as expected
```

Comment:
Please add a test somewhere that checks that "best-so-far" values are returned by each `least_squares` function when it halts prematurely.